nservicebus/upgrades/7to8/index.md

```diff
@@ -249,4 +246,69 @@ if (extensions.TryGet(out DiscardIfNotReceivedBefore constraint))
 {
     timeToBeReceived = constraint.MaxTime;
 }
-```
\ No newline at end of file
+```
+
+## Dependency on System.Memory package for .NET Framework
+
+Memory allocations for incoming and outgoing messages bodies are reduced by using the low allocation memory types via the System.Memory namespace. These type are available on .NET Framework via the **System.Memory** package. The NServiceBus build that targets .NET Framework has this dependency added.
+
+### Pipeline context types changes
+
+Throughput the pipeline all contexts have been updated to use `ReadOnlyMemory<byte>` (e.g. context for behaviors, stage connectors, and message mutators) and affects the following types:
+
+* MutateIncomingTransportMessageContext
+* MutateOutgoingTransportMessageContext
+* ConnectorContextExtensions
+* IIncomingPhysicalMessageContext
+* IncomingPhysicalMessageContext
+* IOutgoingPhysicalMessageContext
+* OutgoingPhysicalMessageContext
+* SerializeMessageConnector
+
+## Message bodies only available in scope of incoming or outgoing messages
+
+Messages bodies available via `ReadOnlyMemory<byte>` are only valid while in scope of incoming or outgoing message processing. Once out of scope the data referenced is no longer valid.
+
+If access is still required outside the processing scope it is required to copy the message body while in scope.
+
+### Message Mutators: Updating of message bodies
+
+The message mutator API for changing the message body has changed. Instead, of `UpdateMessage(byte[] body)` method `MutateIncomingTransportMessageContext` and `MutateOutgoingTransportMessageContext` expose `Body` property of type `ReadOnlyMemory<byte>`.
+
+Message mutators that replace the whole message body might be better of to be converted to a behavior. With a [pipeline behavior](/nservicebus/pipeline/manipulate-with-behaviors.md) it is possible to reduce allocations via [`ArrayPool<byte>`](https://docs.microsoft.com/en-us/dotnet/api/system.buffers.arraypool-1) or packages like [RecyclableMemoryStream](https://github.com/Microsoft/Microsoft.IO.RecyclableMemoryStream).
+
+### Serializers: Deserialization using ReadOnlyMemory of byte
+
+Deserialization updated to use `ReadOnlyMemory<byte>` instead of `Stream` with  on `Deserialize` API method on `IMessageSerializer`.
+
+```csharp
+object[] Deserialize(ReadOnlyMemory<byte> body, IList<Type> messageTypes = null);
+```
+
+If serializers do not support `ReadOnlyMemory<byte>` usage of `.ToArray` must be avoided as this will copy the data and increases memory allocations. Instead include a shim that provides read-only access to the underlying data or use the `ReadOnlyMemoryExtensions.AsStream` from the [Microsoft.Toolkit.HighPerformance](https://docs.microsoft.com/en-us/dotnet/api/microsoft.toolkit.highperformance.extensions.readonlymemoryextensions.asstream).
```

Comment:
```suggestion
Serializers that do not support `ReadOnlyMemory<byte>` as deserialization input should avoid `.ToArray` calls to prevent unnecessary memory allocations. Instead, it's recommended to implement a shim exposing read-only data as types supported by the serializer APIs or use existing implementations like `ReadOnlyMemoryExtensions.AsStream` from the [Microsoft.Toolkit.HighPerformance](https://docs.microsoft.com/en-us/dotnet/api/microsoft.toolkit.highperformance.extensions.readonlymemoryextensions.asstream).
```